pkg/secretsencrypt/controller.go

```diff
@@ -0,0 +1,207 @@
+		return node, err
+	}
+	logrus.Infoln("Removed key: ", curKeys[len(curKeys)-1])
+	if err != nil {
+		h.recorder.Event(node, corev1.EventTypeWarning, secretsUpdateErrorEvent, err.Error())
+		return node, err
+	}
+	if err := WriteEncryptionHashAnnotation(h.controlConfig.Runtime, node, EncryptionReencryptFinished); err != nil {
+		h.recorder.Event(node, corev1.EventTypeWarning, secretsUpdateErrorEvent, err.Error())
+		return node, err
+	}
+	if err := cluster.Save(h.ctx, h.controlConfig, h.controlConfig.Runtime.EtcdConfig, true); err != nil {
+		h.recorder.Event(node, corev1.EventTypeWarning, secretsUpdateErrorEvent, err.Error())
+		return node, err
+	}
+	return node, nil
+}
+
+// validateReencryptStage ensures that the request for reencryption is valid and
+// that there is only one active reencryption at a time
+func (h *handler) validateReencryptStage(node *corev1.Node, annotation string) (bool, error) {
+
+	split := strings.Split(annotation, "-")
+	if len(split) != 2 {
+		err := fmt.Errorf("invalid annotation %s found on node %s", annotation, node.ObjectMeta.Name)
+		return false, err
+	}
+	stage := split[0]
+	hash := split[1]
+
+	// Validate the specific stage and the request via sha256 hash
+	if stage != EncryptionReencryptRequest {
+		return false, nil
+	}
+	if reencryptRequestHash, err := GenReencryptHash(h.controlConfig.Runtime, EncryptionReencryptRequest); err != nil {
+		return false, err
+	} else if reencryptRequestHash != hash {
+		err = fmt.Errorf("invalid hash: %s found on node %s", hash, node.ObjectMeta.Name)
+		return false, err
+	}
+
+	nodes, err := h.nodes.List(metav1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+	reencryptActiveHash, err := GenReencryptHash(h.controlConfig.Runtime, EncryptionReencryptActive)
+	if err != nil {
+		return false, err
+	}
+	for _, node := range nodes.Items {
+		if ann, ok := node.Annotations[EncryptionHashAnnotation]; ok {
+			split := strings.Split(ann, "-")
+			if len(split) != 2 {
+				return false, fmt.Errorf("invalid annotation %s found on node %s", ann, node.ObjectMeta.Name)
+			}
+			stage := split[0]
+			hash := split[1]
+			if stage == EncryptionReencryptActive && hash == reencryptActiveHash {
+				return false, fmt.Errorf("another reencrypt is already active")
+			}
+		}
+	}
+	return true, nil
+}
+
+func (h *handler) updateSecrets(node *corev1.Node) error {
+	secretPager := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
+		return h.secrets.List("", opts)
+	}))
+	i := 0
+	secretPager.EachListItem(h.ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
+		if secret, ok := obj.(*corev1.Secret); ok {
+			if _, err := h.secrets.Update(secret); err != nil {
+				return fmt.Errorf("failed to reencrypted secret: %v", err)
+			}
+			if i != 0 && i%10 == 0 {
+				h.recorder.Eventf(node, corev1.EventTypeNormal, secretsProgressEvent, "reencrypted %d secrets", i)
+			}
+			i += 1
```

Comment:
this appears to be causing a linter warning
```suggestion
			i++
```

Reply:
Fixed